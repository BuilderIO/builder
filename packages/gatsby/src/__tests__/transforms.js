const { buildSchema, printSchema } = require(`gatsby/graphql`)
const {
  NamespaceUnderFieldTransform,
  StripNonQueryTransform,
} = require(`../transforms`)

const sdl = `
  type Query {
    query: String
  }

  type Mutation {
    mutation: String
  }
`

describe(`NamespaceUnderFieldTransform`, () => {
  it(`works`, () => {
    const schema = buildSchema(sdl)

    const transform = new NamespaceUnderFieldTransform({
      typeName: `Wrapper`,
      fieldName: `wrapper`,
      resolver: () => {},
    })
    const transformedSchema = transform.transformSchema(schema)
    const transformedSdl = printSchema(transformedSchema)

    expect(transformedSdl).toMatchSnapshot()
  })
})

describe(`StripNonQueryTransform`, () => {
  it(`works`, () => {
    const schema = buildSchema(sdl)

    const transform = new StripNonQueryTransform()
    const transformedSchema = transform.transformSchema(schema)
    const transformedSdl = printSchema(transformedSchema)

    expect(transformedSdl).toMatchSnapshot()
  })
})
