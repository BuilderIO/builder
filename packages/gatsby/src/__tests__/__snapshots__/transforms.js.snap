// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NamespaceUnderFieldTransform works 1`] = `
"type Mutation {
  mutation: String
}

type Query {
  wrapper: Wrapper!
}

type Wrapper {
  query: String
}
"
`;

exports[`StripNonQueryTransform works 1`] = `
"type Query {
  query: String
}
"
`;
